### ISSUE TITLE
x/pkgsite: @master and @semanticversion show two different things

### ISSUE THREAD

**Issue by SamWhited** [golang/go#43096]
### What did you do?

Visited [https://pkg.go.dev/mellium.im/xmpp/form@master](https://pkg.go.dev/mellium.im/xmpp/form@master) and [https://pkg.go.dev/mellium.im/xmpp/form@v0.17.2-0.20201209031509-c8601a8ec594](https://pkg.go.dev/mellium.im/xmpp/form@v0.17.2-0.20201209031509-c8601a8ec594) which should currently match. I fetched this semantic version by visiting [https://proxy.golang.org/mellium.im/xmpp/@v/master.info](https://proxy.golang.org/mellium.im/xmpp/@v/master.info). The second link works and shows the current package. The first link shows an old version.

I also tried running `go get -u mellium.im/xmpp@master` on a project hoping to force it to update, but after waiting several minutes nothing happened. Either way, if the current default branch version shows up when visiting the semantic version, I would assume it would also be updated when visiting the special `@master` link.

### What did you expect to see?

The documentation should be the same.

### What did you see instead?

The `@master` link shows an old version of the package even though a newer version exists in the database.

**Comment by jamalc** [comment]
Forced a refetch of this module and it looks okay now so I'm going to close.

**Comment by jamalc** [comment]
When a package is visited at @ master we add it to a reprocessing queue. Reprocessing will typically finish in a few minutes but can sometimes take much longer or may be subject to a rate limit.

**Comment by SamWhited** [comment]
I think this is still an issue. The issue is that if the semver of the master version is rendered I would expect master to update immediately too (since the docs have already been generated for that version). It seems like a bug that they show two different things even though it already knows about the new files, and the proxy already knows that this version is the current HEAD on the default branch.

**Comment by jamalc** [comment]
I see what you're saying. I'll leave this open. Adding in some context.

When someone runs `go get -u module@master` for a version the proxy has not seen, the proxy will resolve that to module@semver, module@semver will be added to the new modules feed, and the pkgsite populates its processing queue with module@semver with no understanding of this new version being master.

**Comment by myitcv** [comment]
FYI this remains an issue

At the time of writing, https://pkg.go.dev/cuelang.org/go@master resolves to `v0.3.0-alpha5.0.20201207144701-6de877a926cf`. Whereas the proxy resolves to `v0.3.0-beta.3.0.20210124142225-9c5489fc66ed`:

```
cd $(mktemp -d)
go mod init mod.com
go get cuelang.org/go@master
```

gives:

```
go get: added cuelang.org/go v0.3.0-beta.3.0.20210124142225-9c5489fc66ed
```

cc @julieqiu

**Comment by julieqiu** [comment]
https://pkg.go.dev/cuelang.org/go@master should now be up to date. We're also working on a fix for this general issue.
